app/importers/envirotech/solution_data.rb

```diff
@@ -0,0 +1,61 @@
+module Envirotech
+  class SolutionData
+    include Importer
+    ENDPOINT = 'https://admin.export.gov/admin/envirotech_solutions.json'
+
+
+    COLUMN_HASH = {
+      'id' => :source_id,
+      'name_chinese' => :name_chinese,
+      'name_english' => :name_english,
+      'name_french' => :name_french,
+      'name_portuguese' => :name_portuguese,
+      'name_spanish' => :name_spanish,
+      'created_at' => :source_created_at,
+      'updated_at' => :source_updated_at,
+    }.freeze
+
+    def initialize(resource = ENDPOINT)
+      @resource = resource
+    end
+
+    def import
+      data = fetch_data
+      articles = data.map { |article_hash| process_article_info article_hash }
+      model_class.index articles
+    end
+
+    private
+
+    def fetch_data
+      Envirotech::Login.headless_login
+      result = []
+      result.concat(current_page_data) until current_page_empty?
+      result
+    end
+
+    def current_page_empty?
+      @page ||= 1
+      @page_data = JSON.parse(Envirotech::Login.mechanize_agent.get(@resource + "?page=#{@page}").body)
+      @page += 1
+      @page_data.blank?
+    end
+
+    def current_page_data
+      @page_data
+    end
```

Comment:
>  Side effects in general are a sign of trouble.

Yep, need to improve this, will try your suggestion.


>since this code is repeated in each importer, you should introduce an Envirotech::BaseData class so that common functionality like this can be shared.

Yeah I have this in mind  :+1: